index.js

```diff
@@ -20,11 +20,14 @@ module.exports = class extends URL {
      * @param string base Base URL
      */
     constructor(url, base) {
-        super(url, base);
-
-        // workaround for firefox bug
-        if (this.origin === "null") {
-            Object.defineProperty(this, "origin", {get: () => null});
+        try {
+            super(url, base);
+            Object.defineProperty(this, "isValid", { get: () => true });
```

Comment:
Is there any circumstance under which we would consider something that is parseable by `super()`, to still be an invalid URL?

Reply:
I can't think of anything at the moment 🤔